File: task/src/main/java/com/goexp/galgame/data/db/importor/mongdb/GameDB.java
package com.goexp.galgame.data.db.importor.mongdb;

import com.goexp.common.db.mongo.DBOperatorTemplate;
import com.goexp.common.db.mongo.DBQueryTemplate;
import com.goexp.galgame.data.model.Game;
import org.bson.Document;

import java.util.stream.Collectors;

import static com.mongodb.client.model.Filters.eq;
import static com.mongodb.client.model.Updates.combine;
import static com.mongodb.client.model.Updates.set;

public class GameDB {
    static DBOperatorTemplate tlp = new DBOperatorTemplate("galgame", "game");
    DBQueryTemplate qlp = new DBQueryTemplate<Game>("galgame", "game", doc -> null);

    public void insert(Game game) {

        var gameDoc = new Document("_id", game.id)
                .append("name", game.name)
                .append("publishDate", game.publishDate)
                .append("smallImg", game.smallImg)
                .append("state", 0)
                .append("star", 0)
                .append("state", game.state.getValue())
                .append("brandId", game.brandId);

        tlp.exec(documentMongoCollection -> {
            documentMongoCollection.insertOne(gameDoc);
        });

    }

    public void update(Game game) {

        tlp.exec(documentMongoCollection -> {
            documentMongoCollection.updateOne(eq("_id", game.id),
                    combine(
                            set("publishDate", game.publishDate)
                            , set("smallImg", game.smallImg)
                    ));
        });
    }

    public void updateAll(Game game) {

        tlp.exec(documentMongoCollection -> {
            documentMongoCollection.updateOne(eq("_id", game.id),
                    combine(
                            set("painter", game.painter)
                            , set("writer", game.writer)
                            , set("type", game.type)
                            , set("tag", game.tag)
                            , set("story", game.story)
                            , set("intro", game.intro)
                            , set("brandId", game.brandId)
                    ));
        });
    }

    public void updateChar(Game game) {

        var gameCharDocs = game.gameCharacterList
                .stream()
                .map(gameCharacter -> {
                    var doc = new Document("name", gameCharacter.name)
                            .append("intro", gameCharacter.intro)
                            .append("cv", gameCharacter.cv)
                            .append("truecv", gameCharacter.trueCV)
                            .append("img", gameCharacter.img)
                            .append("index", gameCharacter.index);


                    return doc;

                })
                .collect(Collectors.toList());

        tlp.exec(documentMongoCollection -> {
            documentMongoCollection.updateOne(eq("_id", game.id),
                    set("gamechar", gameCharDocs));
        });
    }

    public void updateImg(Game game) {

        var imgdocs = game.imgList.stream()
                .map(img -> {
                    var doc = new Document("src", img.src)
                            .append("index", img.index);

                    return doc;
                })
                .collect(Collectors.toList());

        tlp.exec(documentMongoCollection -> {
            documentMongoCollection.updateOne(eq("_id", game.id),
                    set("simpleImg", imgdocs));
        });
    }

    public boolean exist(int id) {
        return qlp.exists(eq("_id", id));
    }

    public static class StateDB {
        public void update(Game game) {
            tlp.exec(documentMongoCollection -> {
                documentMongoCollection.updateOne(
                        eq("_id", game.id)
                        , set("state", game.state.getValue())
                );
            });
        }
    }

}

File: task/src/main/java/com/goexp/galgame/data/db/importor/mongdb/GameDB.scala
package com.goexp.galgame.data.db.importor.mongdb

import com.goexp.common.db.mongo.DBOperatorTemplate
import com.goexp.galgame.data.db.query.mongdb.GameQuery
import com.goexp.galgame.data.model.Game
import com.mongodb.client.model.Filters
import com.mongodb.client.model.Updates.{combine, set}
import org.bson.Document

import scala.collection.JavaConverters._


object GameDB {
  var tlp = new DBOperatorTemplate("galgame", "game")

  class StateDB {
    def update(game: Game): Unit =
      tlp.exec(documentMongoCollection => {
        documentMongoCollection.updateOne(Filters.eq(game.id), set("state", game.state.getValue))
      })
  }

}

class GameDB {
  def insert(game: Game): Unit = {
    val gameDoc = new Document("_id", game.id)
      .append("name", game.name)
      .append("publishDate", game.publishDate)
      .append("smallImg", game.smallImg)
      .append("state", 0)
      .append("star", 0)
      .append("state", game.state.getValue)
      .append("brandId", game.brandId)

    GameDB.tlp.exec(documentMongoCollection => {
      documentMongoCollection.insertOne(gameDoc)
    })
  }

  def update(game: Game): Unit =
    GameDB.tlp.exec(documentMongoCollection => {
      documentMongoCollection.updateOne(
        Filters.eq(game.id),
        combine(
          set("publishDate", game.publishDate),
          set("smallImg", game.smallImg)
        )
      )
    })

  def updateAll(game: Game): Unit =
    GameDB.tlp.exec(documentMongoCollection => {
      documentMongoCollection.updateOne(
        Filters.eq(game.id),
        combine(
          set("painter", game.painter),
          set("writer", game.writer),
          set("type", game.`type`),
          set("tag", game.tag),
          set("story", game.story),
          set("intro", game.intro),
          set("brandId", game.brandId))
      )
    })

  def updateChar(game: Game): Unit = {
    val gameCharDocs = game.gameCharacters
      .asScala.toStream
      .map(person => {
        new Document("name", person.name)
          .append("intro", person.intro)
          .append("cv", person.cv)
          .append("truecv", person.trueCV)
          .append("img", person.img)
          .append("index", person.index)
      }).asJava

    GameDB.tlp.exec(documentMongoCollection => {
      documentMongoCollection.updateOne(Filters.eq(game.id), set("gamechar", gameCharDocs))
    })
  }

  def updateImg(game: Game): Unit = {
    val imgdocs = game.gameImgs
      .asScala.toStream
      .map(img => {
        new Document("src", img.src)
          .append("index", img.index)
      }).asJava
    GameDB.tlp.exec(documentMongoCollection => {
      documentMongoCollection.updateOne(Filters.eq(game.id), set("simpleImg", imgdocs))
    })
  }

  def exist(id: Int): Boolean = GameQuery.fullTlp.query.where(Filters.eq(id)).exists
}